### PR TITLE
refactor: remove depedency on lock factory

### PR DESCRIPTION
**Pull request by saranyailla**
**Issue #, if available:**

**Description of changes:**
PR for removing dependency on cycle detection (that was introduced in nucleus 2.11.0) in Secret manager. This will let secret manager have dependency on nucleus >=2.5.0

**Why is this change necessary:**

**How was this change tested:**

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java [hunk 1]
```diff
@@ -55,10 +51,8 @@ public class SecretManagerService extends PluginService {
 
     private final SecretManager secretManager;
     private final AuthorizationHandler authorizationHandler;
-    private final ExecutorService executor;
```
Comment (junfuchen99) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:58]: why removing this?

**Review by junfuchen99 — APPROVED** [review]

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java (hunk 1 above)
Comment (saranyailla) [reply src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java:58]: It was never used in the code.

**Review by alter-mage — APPROVED** [review]

**Review by junfuchen99 — APPROVED** [review]

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java [hunk 2]
```diff
@@ -73,6 +73,7 @@ public void setCloudCallThreadPool(Topics config) {
                 .findOrDefault(DEFAULT_CLOUD_REQUEST_SIZE, CLOUD_REQUEST_QUEUE_SIZE_TOPIC));
         if (cloudCallQueueSize <= 0) {
             logger.atWarn().kv("queueSize", DEFAULT_CLOUD_REQUEST_SIZE)
+                    .kv("configured", cloudCallQueueSize)
```
Comment (alter-mage) [inline src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:76]: is this a development change or actually something we want to release?

**Review by alter-mage — APPROVED** [review]

File: src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java (hunk 2 above)
Comment (saranyailla) [reply src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java:76]: Yes, we can keep it. The previous log message doesn't show the configured value.

**Pull request merged**